optd-cost-model/src/cost_model.rs

```diff
@@ -40,28 +41,83 @@ impl<S: CostModelStorageManager> CostModelImpl<S> {
     }
 }
 
+#[async_trait::async_trait]
 impl<S: CostModelStorageManager + Send + Sync + 'static> CostModel for CostModelImpl<S> {
-    fn compute_operation_cost(
+    async fn compute_operation_cost(
         &self,
         node: &PhysicalNodeType,
         predicates: &[ArcPredicateNode],
-        children_stats: &[Option<&EstimatedStatistic>],
+        children_stats: &[EstimatedStatistic],
         context: ComputeCostContext,
     ) -> CostModelResult<Cost> {
         todo!()
     }
 
-    fn derive_statistics(
+    async fn derive_statistics(
```

Comment:
Make this pub?

Reply:
The methods are public because `trait CostModel` is public.

Reply:
Oh you're right